### PR TITLE
feat: Check for updates

### PR DESCRIPTION
**Pull request by ashchan**
Add menu item allowing user to check updates.

### CODE REVIEW

**Comment by Keith-CY** [comment]
CI failed

**Comment by ashchan** [comment]
> CI failed

Checking that

File: packages/neuron-wallet/src/controllers/app/index.ts [hunk 1]
```diff
@@ -83,6 +87,9 @@ export default class AppController {
       connectionStatus,
       codeHash,
     }
+
+    AppController.handleWalletListChange()
```
Comment (Keith-CY) [inline packages/neuron-wallet/src/controllers/app/index.ts]: `getInitState` would be called multiple times, which makes the subscription repeated.

File: packages/neuron-wallet/src/controllers/app/index.ts [hunk 2]
```diff
@@ -155,6 +162,10 @@ export default class AppController {
     AppController.showMessageBox(options)
   }
 
+  public static checkUpdates(menuItem: MenuItem) {
+    UpdateController.checkUpdates(menuItem)
```
Comment (Keith-CY) [inline packages/neuron-wallet/src/controllers/app/index.ts]: The name `update controller checks updates menu item` is confusing.

Comment (ashchan) [reply packages/neuron-wallet/src/controllers/app/index.ts]: Indeed. What's your suggestion? We do need to pass the menu item for disabling/enabling.

File: packages/neuron-wallet/src/controllers/app/index.ts (hunk 1 above)
Comment (ashchan) [reply packages/neuron-wallet/src/controllers/app/index.ts]: Where's the appropriate place to initialize this?

File: packages/neuron-wallet/src/controllers/app/index.ts (hunk 2 above)
Comment (ashchan) [reply packages/neuron-wallet/src/controllers/app/index.ts]: `checkUpdates(sender: MenuItem)`, `checkUpdates(sender: any)`?

Comment (Keith-CY) [reply packages/neuron-wallet/src/controllers/app/index.ts]: If [Menu.getApplicationMenu](https://electronjs.org/docs/api/menu#menugetapplicationmenu) works in the update controller, the `menuitem` parameter is redundant.

Comment (ashchan) [reply packages/neuron-wallet/src/controllers/app/index.ts]: We might trigger this by other UI component, eg from a button. `sender: any` is a common signature when programming macOS/iOS app.

Comment (ashchan) [reply packages/neuron-wallet/src/controllers/app/index.ts]: > If [Menu.getApplicationMenu](https://electronjs.org/docs/api/menu#menugetapplicationmenu) works in the update controller, the `menuitem` parameter is redundant.

Not exactly. We can't be sure the menu item triggering it is from the main menu. (But if we have multiple entries then we do need to disable/enable all).

I'd say we leave this as. If we add more action entries then we refactor/rename this.

Comment (Keith-CY) [reply packages/neuron-wallet/src/controllers/app/index.ts]: LGTM

File: packages/neuron-wallet/src/controllers/app/index.ts (hunk 1 above)
Comment (Keith-CY) [reply packages/neuron-wallet/src/controllers/app/index.ts]: We can subscribe to the subjects just on the app launched.

Comment (ashchan) [reply packages/neuron-wallet/src/controllers/app/index.ts]: I see. Thanks.

Comment (ashchan) [reply packages/neuron-wallet/src/controllers/app/index.ts]: @Keith-CY Moved that to `app.ready` event block. Please let me know if there's a better launch entry.

**Pull request merged**